util/contentutil/copy.go

```diff
@@ -81,7 +84,8 @@ func CopyChain(ctx context.Context, ingester content.Ingester, provider content.
 		retryhandler.New(limited.FetchHandler(ingester, &localFetcher{provider}, ""), func(_ []byte) {}),
 	}
 
-	if err := images.Dispatch(ctx, images.Handlers(handlers...), nil, desc); err != nil {
+	ctx2 := remotes.WithMediaTypeKeyPrefix(ctx, intoto.PayloadType, "intoto")
```

Comment:
Although you want to be precise is here and only change the context where it is needed it is better if this is always the first line of the function because the call is not specific to "intoto" at all. Also add a comment where this is used that this registers types that are not defined by default. Otherwise this reads like `push.Push()` or `contentutil.Copy` always register current payloadtype as "intoto".

Maybe this is better in a util pkg function, eg. `RegisterContentPayloadTypes()`

Reply:
> Although you want to be precise is here and only change the context where it is needed it is better if this is always the first line of the function because the call is not specific to "intoto" at all

Makes sense, it's a lot cleaner all at the top :smile:

> Maybe this is better in a util pkg function, eg. RegisterContentPayloadTypes()

Agreed, have moved this to `contentutil.RegisterContentPayloadTypes()`.